Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift

```diff
@@ -131,21 +125,21 @@ struct ComboBoxInput: View {
         }
         .padding([.bottom], elementPadding)
         .onAppear {
-            codedValues = featureForm!.codedValues(fieldName: element.fieldName)
-            selectedValue = codedValues.first { $0.name == element.value }
+            codedValues = model.featureForm!.codedValues(fieldName: element.fieldName)
+            selectedValue = codedValues.first { $0.name == inputModel.formattedValue }
         }
-        .onChange(of: selectedValue) { newValue in
-            guard newValue?.name != inputModel.value else {
-                return
+        .onChange(of: selectedValue) { selectedValue in
+            requiredValueMissing = inputModel.isRequired && selectedValue == nil
+            do {
+                try element.updateValue(selectedValue?.code)
+            } catch {
+                print(error.localizedDescription)
             }
-
-            requiredValueMissing = element.isRequired && newValue == nil
-            featureForm?.feature.setAttributeValue(newValue?.code, forKey: element.fieldName)
             model.evaluateExpressions()
         }
-        .onChange(of: inputModel.value) { newValue in
-            let codedValues = featureForm!.codedValues(fieldName: element.fieldName)
-            selectedValue = codedValues.first { $0.name == newValue }
+        .onChange(of: inputModel.formattedValue) { _ in
+            let codedValues = model.featureForm!.codedValues(fieldName: element.fieldName)
+            selectedValue = codedValues.first { $0.name == inputModel.formattedValue }
```

Comment:
This doesn't cause an infinite loop with the `.onChange(of: selectedValue)`, does it?

Reply:
From what I've seen so far, no.